### PR TITLE
Thieving Stalls

### PR DESCRIPTION
**Pull request by Lajnux**
### Description:

Added stalls as stealable objects and balanced rates to be similar to in game along with requirements.

### Changes:

New Stalls as stealable objects

### Other checks:

-   [X] I have tested all my changes thoroughly.

### CODE REVIEW

File: src/commands/Minion/pickpocket.ts [hunk 1]
```diff
@@ -63,91 +67,128 @@ export default class extends BotCommand {
...
+		if (!hasFavour && stealable.name === 'Fruit stall') {
+			return msg.channel.send(
+				`${msg.author.minionName} needs ${requiredPoints}% Hosidius Favour to steal fruit from the Fruit stalls!`
+			);
+		}
 
-		const maxTripLength = msg.author.maxTripLength('Pickpocket');
+		const timeToTheft = isPickpockable
+			? ((<Pickpockable>stealable).customTickRate ?? 2) * 600
+			: (<Stall>stealable).respawnTime;
```
Comment (gc) [inline src/commands/Minion/pickpocket.ts:128]: we dont do stuff like this anywhere. you should convey this in the type itself. rough example:
```ts
type Stealable = { type: 'pickpocket' | 'stall' } & (PickpocketNPC | Stall);
type Pickpocketable {
type: 'pickpocket',
customTickRate: 1,
}
type Stall {
type: 'stall'
}

const timeToTheft = stealable.type === 'pickpocket' ? stealable.customTickrate : stealable.whatever;
```

**Review by gc — CHANGES_REQUESTED** [review]
1. the 1 comment i left
2. we should make it into a centralized `steal` slash command, which accepts any npc/stall.

After these 2 things, and assuming its tested and the rates are measured/balanced, this looks good to merge.

File: src/lib/skilling/skills/thieving/stealables.ts [hunk 2]
```diff
@@ -64,34 +63,46 @@ export const Stalls: Stall[] = [
...
 			.tertiary(206_777, 'Rocky'),
 		qpRequired: 3,
-		respawnTime: Time.Second * 2
+		respawnTime: Time.Second * 2,
+		lootPercent: 20
 	},
 	{
 		name: "Baker's stall",
+		type: 'stall',
+		alias: ['Baker stall', 'Baker', 'Bakers stall'],
```
Comment (gc) [inline src/lib/skilling/skills/thieving/stealables.ts:72]: aliases, in the bot and osjs, we always have them all lowercase

File: src/lib/skilling/skills/thieving/stealables.ts [hunk 3]
```diff
@@ -28,31 +28,30 @@ import { ItemBank } from '../../../types';
 
 export interface Stealable {
 	name: string;
+	type: 'pickpockable' | 'stall';
+	alias?: string[];
```
Comment (gc) [inline src/lib/skilling/skills/thieving/stealables.ts:32]: `aliases`

**Review by gc — CHANGES_REQUESTED** [review]
also have you checked xp rates of everything?

**Comment by Lajnux** [comment]
> also have you checked xp rates of everything?

yupp, it's all good and fair and tested, had a extra test up now today aswell.

File: src/mahoji/commands/steal.ts [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+		const [hasFavour, requiredPoints] = gotFavour(user, Favours.Hosidius, 15);
+		if (!hasFavour && stealable.name === 'Fruit stall') {
+			return `${user.minionName} needs ${requiredPoints}% Hosidius Favour to steal fruit from the Fruit stalls!`;
+		}
+
+		const timeToTheft =
+			stealable.type === 'pickpockable' ? (stealable.customTickRate ?? 2) * 600 : stealable.respawnTime;
+
+		if (!timeToTheft) {
+			return 'This NPC/Stall is not working as intended, report it.';
```
Comment (gc) [inline src/mahoji/commands/steal.ts:94]: dont do this - use `logError` so we can actually know if something like this happens

**Review by gc — CHANGES_REQUESTED** [review]

**Pull request merged**